installer: Improve status reporting

michaelkuhn opened the pull request:
Refactor `TermTitle` into `InstallStatus` and use it to show progress information both in the terminal title as well as inline. This also turns on the terminal title status by default.

The inline output will look like the following after this change:
```
==> Installing m4-1.4.19-w2fxrpuz64zdq63woprqfxxzc3tzu7p3 [4/4]
```

cc @haampie @alalazo I mentioned this change during the BoF.

michaelkuhn committed:
installer: Improve status reporting

Refactor `TermTitle` into `InstallStatus` and use it to show progress
information both in the terminal title as well as inline. This also
turns on the terminal title status by default.

The inline output will look like the following after this change:
```
==> Installing m4-1.4.19-w2fxrpuz64zdq63woprqfxxzc3tzu7p3 [4/4]
```

alalazo commented:
How does this play with depfiles (sorry, didn't have time to try it myself yet)?

michaelkuhn commented:
The terminal title status also does not do anything meaningful with depfiles (mostly `[1/1]`). Should we toggle it off for depfiles? I guess it would be pretty hard to make it work with multiple Spack processes.

alalazo commented:
> Should we toggle it off for depfiles? 

That would be fine with me. Ping me as soon as you do and I'll merge this one.

michaelkuhn committed:
templates: Do not show install status

Due to each package being installed separately, status information would
mostly show `[1/1]`.

michaelkuhn commented:
@alalazo ping :slightly_smiling_face: I've renamed the config option to `install_status`, which controls both the inline and terminal title status information. For depfiles, we simply toggle it off using `spack -c`.

alalazo commented:
@spackbot run pipeline

alalazo commented:
@becker33 @tgamblin Do we need to deprecate the old name, or can we just change it for this option?

michaelkuhn commented:
@alalazo I've rebased this to get rid of the conflicts. I also have a local change to also check the old config name (in an effort to deprecate it) but I'm not sure how much good it actually does: If people enabled `terminal_title` explicitly, they will also get the new output (since it's enabled by default). If they never changed it but want to keep it disabled, they will have to look at the config anyway since this PR turns it on.

alalazo commented on lib/spack/spack/schema/config.py:
```diff
@@ -87,6 +87,7 @@
                 "anyOf": [{"type": "integer", "minimum": 1}, {"type": "null"}]
             },
             "allow_sgid": {"type": "boolean"},
+            "install_status": {"type": "boolean"},
```
I would just substitute "module_roots" below at line 96 with `terminal_title`  and say it has no effect

michaelkuhn replied:
Thanks for the hint, I completely missed this. I just pushed a new commit that replaces the old message with a new one for `terminal_title` (I hope that is what you meant).

michaelkuhn committed:
config: Deprecate terminal_title

alalazo commented:
Thanks!
